app/src/main/java/com/capstone/weatherapp/WeatherRepository.kt

```diff
@@ -10,11 +10,11 @@ private const val APIKEY = "6849ae760f417fb8188f4bb7fd0d92fc"
 private const val UNITS = "imperial"
 
 class WeatherRepository(private val db: WeatherDatabase) {
-    private val _cityListData = MutableLiveData<List<City>>()
-    val cityListData: LiveData<List<City>>
+    private val _cityListData = MutableLiveData<State<List<City>>>()
```

Comment:
once you fix the above comment about your State classes, this should look much cleaner. Since you won't need to specify any parameters for State, it should look something like this once updated:

`private var _cityListData: MutableLiveData<GroupDataState> = MutableLiveData(GroupDataState.Loading)`